pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -106,6 +107,12 @@ type WorkflowList struct {
 var _ TemplateGetter = &Workflow{}
 var _ TemplateStorage = &Workflow{}
 
+// TTLStrategy is the strategy for the time to live depending on if the workflow succeded or failed
+type TTLStrategy struct {
+	SecondsAfterSuccess *int32 `json:"SecondsAfterSuccess:,omitempty" protobuf:"bytes,18,opt,name=SecondsAfterSuccess:"`
```

Comment:
Our convention has `json` variables be `camelCase`. Also, start with a fresh `protobuf` index for each `struct`
```suggestion
	SecondsAfterSuccess *int32 `json:"secondsAfterSuccess:,omitempty" protobuf:"bytes,1,opt,name=SecondsAfterSuccess:"`
```